spec/jobs/allocation_job_spec.rb

```diff
@@ -279,33 +338,102 @@
         expect(school1.std_device_allocation.raw_cap).to eql(school1.std_device_allocation.raw_allocation)
       end
 
+      context 'updating to cannot_order' do
+        let(:batch_allocation_job1) do
+          create(:allocation_batch_job, urn: school1.urn, allocation_delta: '100', order_state: 'cannot_order')
+        end
+        let(:batch_allocation_job2) do
+          create(:allocation_batch_job, urn: school2.urn, allocation_delta: '100', order_state: 'cannot_order')
+        end
+
+        it 'updates the raw_cap to match raw_devices_ordered' do
+          described_class.perform_now(batch_allocation_job1)
+          described_class.perform_now(batch_allocation_job2)
+
+          expect(school1.std_device_allocation.raw_cap).to eql(school1.std_device_allocation.raw_devices_ordered)
+          expect(school2.std_device_allocation.raw_cap).to eql(school2.std_device_allocation.raw_devices_ordered)
+        end
+      end
+
+      context 'updating to cannot_order with a zero delta' do
+        let(:batch_allocation_job1) do
+          create(:allocation_batch_job, urn: school1.urn, allocation_delta: '0', order_state: 'cannot_order')
+        end
+        let(:batch_allocation_job2) do
+          create(:allocation_batch_job, urn: school2.urn, allocation_delta: '0', order_state: 'cannot_order')
+        end
+
+        it 'does not change the allocation' do
+          expect {
+            described_class.perform_now(batch_allocation_job1)
+          }.to change { school1.std_device_allocation.reload.allocation }.by(0)
```

Comment:
Don't worry about this now but you can write lines like this as `}.to_not change { school1.std_device_allocation.reload.allocation }`

Reply:
Easy change. Done!